api/src/pcapi/routes/backoffice/filters.py

```diff
@@ -833,6 +833,12 @@ def format_finance_incident_type(incident_kind: finance_models.IncidentType) ->
             return incident_kind.value
 
 
+def format_quantity_stock(quantity: typing.Optional[int]) -> int | str:
+    if quantity is None:
+        return "Illimité"
+    return quantity
```

Comment:
Je suis plutôt d'avis de renvoyer toujours une `str`, puisque c'est comme pour un `print` dans le html, ça évite deux types de retour.

```suggestion
def format_quantity_stock(quantity: int | None) -> str:
    if quantity is None:
        return "Illimité"
    return str(quantity)
```
Sinon, `format_stock_quantity` me semblerait un nommage plus homogène avec l'existant.